src/cnn/mfcc_model.py

```diff
@@ -0,0 +1,69 @@
+import torch
+import torch.nn as nn
+
+class MFCC_CNN(nn.Module):
+    def __init__(self):
+        super(MFCC_CNN, self).__init__()
+        self.name = "mfcc_cnn"
+
+        self.conv1 = nn.Conv2d(3, 16, kernel_size=5, stride=1, padding=1)
+        self.relu1 = nn.ReLU()
+        self.maxpool1 = nn.MaxPool2d(kernel_size=2, stride=2)
+
+        self.conv2 = nn.Conv2d(16, 32, kernel_size=3, stride=1, padding=0)
+        self.relu2 = nn.ReLU()
+        self.maxpool2 = nn.MaxPool2d(kernel_size=2, stride=2)
+
+        self.conv3 = nn.Conv2d(32, 64, kernel_size=3, stride=1, padding=0)
+        self.relu3 = nn.ReLU()
+        self.maxpool3 = nn.MaxPool2d(kernel_size=2, stride=2)
+
+        self.conv4 = nn.Conv2d(64, 128, kernel_size=3, stride=1, padding=0)
+        self.relu4 = nn.ReLU()
+        self.maxpool4 = nn.MaxPool2d(kernel_size=2, stride=2)
+
+        # Calculate the size of the input tensor to the fully connected layer
+        self.fc_input_size = self.calculate_fc_input_size()
+
+        self.fc1 = nn.Linear(self.fc_input_size, 512)
+        self.relu5 = nn.ReLU()
+        self.fc2 = nn.Linear(512, 10)
+
+    def calculate_fc_input_size(self):
+        # Perform a forward pass to get the output shape of the last convolutional layer
```

Comment:
Wouldn't it be more efficient/better practice to use the output formulas we've learned in class to determine the output size of the last convolutional layer

Reply:
yes it would be. This formula makes my life easier when I was messing around with the convolution settings / layers. And in terms of efficiency this is only called once during initialization, so it isn't too big of a performance hit. Does it help with your concern?

Reply:
Yes, good point about it only being called once👍🏽